src/types/products.ts

```diff
@@ -20,7 +20,14 @@ type ProductName =
 	| 'Vault'
 	| 'Waypoint'
 
-type ProductSlug = Exclude<Products, 'hashicorp'> | 'hcp' | 'sentinel'
+type ProductSlug =
+	| Exclude<Products, 'hashicorp'>
+	| 'hcp'
+	| 'sentinel'
+	// This enables literals (e.g. 'boundary') in conjuction with a generic `string` type
+	// `(string & {})` is a workaround for this issue: https://github.com/Microsoft/TypeScript/issues/29729
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	| (string & {})
```

Comment:
🙋🏻 Is this change directly related to the change in the other file? I'm not sure I understand how this could implicate our _many_ references to `ProductSlug`

Reply:
Yes, it's to accommodate for the improved typings introduced to `__config`, in `global.d.ts`.
That change ultimately replaced an `any` with `string`, which caused this line to have a type error:
- https://github.com/hashicorp/dev-portal/blob/kw.config-intellisense/src/components/command-bar/commands/search/components/integrations-tab-contents/index.tsx#L23

Reply:
Oh okay cool! That's helpful context. I'm going to check out the code locally real quick, that'll help me visualize these changes. Brb! 😊

Reply:
Have moved some discussion "offline" to Slack in case it's faster!